File: src/PIMTool/Database/Project.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.RegularExpressions;

namespace PIMTool.Database
{
    public class Project
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public int ProjectNumber { get; set; }

        [Required, MaxLength(50)]
        public string Name { get; set; }

        [Required, MaxLength(50)]
        public string Customer { get; set; }

        [Required, MaxLength(3)]
        public string Status { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        [Required]
        public int Version { get; set; }

        public int GroupId { get; set; }

        public Group Group { get; set; }

        public ICollection<ProjectEmployee> ProjectEmployees { get; set; }

    }
}

File: PimToolFE/src/app/core/features/project/models/project.ts
export interface Project {
  id?: number
  projectNumber?: number;
  name?: string;
  customer?: string;
  groupId?: number;
  employeeVisas?: string[];
  status?: string;
  startDate?: string | null;
  endDate?: string | null;
  totalPage?: number;
  version?: Uint8Array;
}
